STYLE.md

```diff
@@ -9,6 +9,26 @@ particular for proto3 as described at:
 
 In addition, the following conventions should be followed:
 
+* For protos that are [frozen](https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/v2_overview#status),
+  the following guidelines are followed:
+
+  * Fields should never be re-indexed. This is standard proto development procedure.
```

Comment:
Maybe express it as "Fields should not be renumbered or have their types changed"